src/saving/SaveHelper.cs

```diff
@@ -131,11 +131,89 @@ private static Save CreateSaveObject(MainGameState gameState, SaveInformation.Sa
         };
     }
 
+    private static string BuildQuicksaveFilename(uint quicksaveId)
+    {
+        return $"quick_save_{quicksaveId}.{Constants.SAVE_EXTENSION}";
+    }
+
+    /// <summary>
+    ///   Returns the next quicksave-id to be used
+    /// </summary>
+    private static uint GetNextQuicksaveId()
+    {
+        var files = GetSaves();
+        for (uint i = 0; i < Constants.SAVE_MAX_QUICKSAVES_BEFORE_OVERRIDING_OLD_ONES; i++)
+        {
+            if (!files.Contains(BuildQuicksaveFilename(i)))
+            {
+                // Not all 5 quicksaves have been used
+                return i;
+            }
+        }
+
+        // All 5 quicksaves have been used
+        var filename = GetLastSavegame(true);
+        return (uint.Parse(
+                        filename.Substr("quick_save_".Length,
+                        filename.Length - Constants.SAVE_EXTENSION.Length - "quick_save_".Length - 1),
+                        System.Globalization.CultureInfo.InvariantCulture)
+                    + 1) % (Constants.SAVE_MAX_QUICKSAVES_BEFORE_OVERRIDING_OLD_ONES + 1);
+    }
+
+    private static string GetLastModifiedFile(List<string> filesToCheck)
+    {
+        return filesToCheck.ToDictionary(p => p, p =>
+        {
+            using (var file = new File())
+            {
+                return file.GetModifiedTime(PathUtils.Join(Constants.SAVE_FOLDER, p));
+            }
+        }).Aggregate((a, b) => a.Value > b.Value ? a : b).Key;
```

Comment:
But aggregate is faster than sorting. And it makes more sense to have the filename as the key and the modified date as the value, not the other way around.

Reply:
I think that code like this would work (I didn't test or write with C# support enabled so this might not compile):
```
using (var file = new File())
{
    return filesToCheck.SortBy(entry => file.GetModifiedTime(PathUtils.Join(Constants.SAVE_FOLDER, p))).First();
}
```

Which, in my opinion is much nicer than making a temporary dictionary and recreating the file object for each file.